common/sgx_workload/trusted_iohandler.cpp

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <stdlib.h>
+#include <stdio.h>
+#include <iostream>
+#include <stdint.h>
+
+#include "trusted_iohandler.h"
+#include "iohandler_enclave.h"
+#include "enclave_t.h"
+#include "enclave_utils.h"
+
+uint32_t GetIoHandlerId(const char* handlerName)
+{
+    return TcfGetIoHandlerId(handlerName);
+}
+
+uint32_t FileOpen(uint32_t handlerId, std::string fileName, uint8_t *result,
+    size_t resultSize)
+{
+	uint32_t status;
+	std::string fileOperation = "open";
+	std::string command = fileOperation + " " + fileName;
+	status = TcfExecuteIoCommand(handlerId, (const uint8_t *)command.c_str(),
+            command.length(), result, resultSize, NULL, 0, NULL, 0);
+	return status;
+}
+
+uint32_t FileClose(uint32_t handlerId, std::string fileName, uint8_t *result,
+    size_t resultSize)
+{
+	uint32_t status;
+	std::string fileOperation = "close";
+	std::string command = fileOperation + " " + fileName;
+	status = TcfExecuteIoCommand(handlerId, (const uint8_t *)command.c_str(),
+            command.length(), result, resultSize, NULL, 0, NULL, 0);
+	return status;
+}
+
+uint32_t FileRead(uint32_t handlerId, std::string fileName, uint8_t *result,
+    size_t resultSize, uint8_t *outBuf, size_t outBufSize)
```

Comment:
Remove `outBufSize` here and everywhere else. It is not used.

Reply:
OutBufSize can be used to compare whether size of the output buffer content is matched with the outBufSize.

Reply:
For C strings you only need to know the buffer size if the C string size could be increased in the function. But for these functions, `const uint8_t *outBuf` is a read-only string and will not be modified (and will not have an overflow).  The length check is not needed.

Furthermore, the function is called passing a C++ string converted to a C string.  The inBufSize will always be the length of the string.

Reply:
outBuf is not read only. content read from the file is copied to outBuf, hence we need the size.